vortex-ipc/src/io/monoio.rs

```diff
@@ -6,13 +6,14 @@ use std::io;
 use bytes::BytesMut;
 use futures_util::FutureExt;
 use monoio::buf::IoBufMut;
-use monoio::io::{AsyncReadRent, AsyncReadRentExt};
+use monoio::io::{AsyncReadRent, AsyncReadRentExt, AsyncWriteRent, AsyncWriteRentExt};
+use vortex_buffer::io_buf::IoBuf;
 
-use crate::io::VortexRead;
+use crate::io::{VortexRead, VortexWrite};
 
-pub struct MonoVortexRead<R: AsyncReadRent>(R);
+pub struct MonoAdapter<IO>(IO);
```

Comment:
It's weird that this is the adapter for buffers as well as an adapter for read/write operations?

Reply:
i.e. we use it for orthogonal features but it's just an adapter...